### PR TITLE
Reject bad epochs in apply_ica diagnostic plot

### PR DESCRIPTION
**Pull request by crsegerie**
Following a discussion with @agramfort, @SophieHerbst and @hoechenberger, there were indeed some inconsistencies between the calculation of the ica and the plotting of the correction of the ica in the short ica report.

See, for this case study, before:

image

and after:

image

### Before merging …

- [x] Changelog has been updated (`docs/source/changes.md`)

### CODE REVIEW

File: docs/source/changes.md [hunk 1]
```diff
@@ -120,3 +120,7 @@ authors:
 - `epochs_tmin` and `epochs_tmax` were named incorrectly in some test config
   files.
   ({{ gh(340) }} by {{ authors.crsegerie }})
+- We now reject bad epochs trespassing `ica_reject` before plotting the
```
Comment (agramfort) [inline docs/source/changes.md:123]:
```suggestion
- We now reject bad epochs by using `ica_reject` before plotting the 
```

**Review by agramfort — APPROVED** [review]

File: docs/source/changes.md [hunk 2]
```diff
@@ -120,3 +120,7 @@ authors:
 - `epochs_tmin` and `epochs_tmax` were named incorrectly in some test config
   files.
   ({{ gh(340) }} by {{ authors.crsegerie }})
+- We now reject bad epochs trespassing `ica_reject` before plotting the 
+  figures in the `proc-ica_report`.
```
Comment (hoechenberger) [inline docs/source/changes.md]: Haha, nicely phrased, I giggled :)

But I'm not sure if it's easy to understand for someone who wasn't involved in our discussion to understand what has been fixed here. Proposal:

```suggestion
- When creating the "overlay" plot to show the effects of ICA cleaning, we
  previously plotted the data without applying the pre-ICA peak-to-peak
  rejection thresholds, producing misleading figures in some situations. This
  has now been fixed.
```

File: docs/source/changes.md [hunk 3]
```diff
@@ -120,3 +120,7 @@ authors:
 - `epochs_tmin` and `epochs_tmax` were named incorrectly in some test config
   files.
   ({{ gh(340) }} by {{ authors.crsegerie }})
+- We now reject bad epochs by using `ica_reject` before plotting the
```
Comment (hoechenberger) [inline docs/source/changes.md:123]: Can you insert a link to the docs here? 
```
[`ica_reject`][config....]
```

File: docs/source/changes.md [hunk 4]
```diff
@@ -120,3 +120,7 @@ authors:
 - `epochs_tmin` and `epochs_tmax` were named incorrectly in some test config
   files.
   ({{ gh(340) }} by {{ authors.crsegerie }})
+- We now reject bad epochs by using `ica_reject` before plotting the 
+  figures in the `proc-ica_report`.
+  ({{ gh(?) }} by {{ authors.hoechenberger }}), {{ authors.agramfort }} and 
+  {{ authors.crsegerie }})
```
Comment (hoechenberger) [inline docs/source/changes.md:126]:
```suggestion
  ({{ gh(385) }} by {{ authors.crsegerie }})
```
It's your PR :)

File: scripts/preprocessing/05a-apply_ica.py [hunk 5]
```diff
@@ -57,6 +57,8 @@ def apply_ica(cfg, subject, session):
     # Load epochs to reject ICA components.
     epochs = mne.read_epochs(fname_epo_in, preload=True)
 
+    epochs.drop_bad(cfg.ica_reject)
```
Comment (hoechenberger) [inline scripts/preprocessing/05a-apply_ica.py:60]:
```suggestion
    epochs.drop_bad(cfg.ica_reject)
```

File: docs/source/changes.md (hunk 2 above)
Comment (hoechenberger) [reply docs/source/changes.md]: I see you've meanwhile updated the changelog, so ignore this comment

File: docs/source/changes.md (hunk 3 above)
Comment (hoechenberger) [reply docs/source/changes.md:123]: Proposal:

"… before producing the "overlay" plots that show the evoked data before and after ICA cleaning."

Comment (crsegerie) [reply docs/source/changes.md:123]: ok!

**Review by hoechenberger — APPROVED** [review]

**Comment by hoechenberger** [comment]
Seems OpenNeuro is having some issues again …

**Comment by hoechenberger** [comment]
Thanks @crsegerie!

**Pull request merged**